### DISCUSSION TITLE
I tried to split my date form, without success!

### DISCUSSION THREAD

**Discussion by migueljcpinto** [neuefische/questions#701] [Web] [answered]
In my "WorkScheduleForm" form, I have two different Datepickers for two types of dates (Holidays - Date Range; Days off - Multiple days) and the form was submitted with the result of the two Datepickers, generating errors and empty arrays if, for example, no day was selected in Days off. That's why I thought of splitting the form with two submit functions for each Datepicker. However, it didn't work.

The data consists of vacation dates (vacationDates) and days off (daysOff). When submitting vacation dates or days off through the frontend, I receive a 400 Bad Request error. According to the console.logs, indicate that the data is being received correctly on the server-side but not saved in DB. 

The NurseWorkDates model in the database is designed to store both vacationDates (an array of date ranges) and daysOff (an array of individual dates). The problem may be on validation and data processing logic in the API routes but I can not see where or how.

[PR](https://github.com/MiketheBite/Capstone-project/pull/15)

**Comment by F-Kirchhoff** [comment]
Hey Miguel,
I guess the problem is that this task is a bit convoluted. You can try to create two separate API endpoints, for vacation and daysOff respectively. Then you need to either
- get the corresponding work-dates for the nurse if it already exists and add the new data to it
- change your work-dates schema, such that it can hold an array of days And a "type" which can be either vacation or daysOff. Then you can add a new work-date document for each new entry of the nurse and don't have to worry to find the old document and update it but rather just create another one. In the frontend you need then to find all vacations / daysOff of the corresponding nurse.
Hope that helped 😊

**Reply by migueljcpinto** [reply]
Thank you for your answer🙏, that really clarifies the question a bit, although I still need to see how should I do it. Considering that I still want to add the availability of work for several days through three different shifts, maybe adding "types" could be the best option, right?

**Reply by F-Kirchhoff** [reply]
I think thats the best way too. But you should add another "time" key for the different shifts I guess instead of using the "type" as well for that.

**Reply by F-Kirchhoff** [reply, marked as answer]
I would suggest you refactor your data structure, such that you have one collection of "absence" documents. In these, one nurse can have multiple absences, referenced by their nurseID. When you want to display the absences of the specific nurse in the frontend, you need to find all absences in this collection with the respective nurseID and then display them as a list. The absences can have two types, "dayOff" and "vacation" and I would suggest that you create a new document for every single "dayOff" absence, since they can be chosen one by one in your input form. I hope this gives you enough guideline for you to continue 😊
